src/engine/framework/BaseCommands.cpp

```diff
@@ -113,13 +113,13 @@ namespace Cmd {
                 int filenameArg = hasOptions ? 2 : 1;
                 const std::string& filename = args.Argv(filenameArg);
 
-                SetExecArgs(args, filenameArg + 1);
                 if (not ExecFile(filename, executeSilent)) {
                     if (not failSilent) {
                         Print("couldn't exec '%s'", filename.c_str());
                     }
                     return;
                 }
+                SetExecArgs(args, filenameArg + 1);
```

Comment:
This is super confusing.  At the very least this deserves a comment. Better yet, we should update the implementations of both of things to use `BufferCommandText` so it behaves like we'd expect the execution to behave.

Reply:
Commented. Not sure what you mean by the last sentence. I think just using Cvar::SetValue instead of commands would also work here if that seems better

Reply:
I mean using `Cmd::BufferCommandText` directly instead of ExecuteAfter, which is the root of this confusing behavior. If you use the former method, it will queue up commands as expected.

Would be interesting to see that. However, using `Cvar::SetValue` is certainly more clear in terms of intent instead of relying on the command system to set the cvars.

Reply:
Switched to Cvar::SetValue.

Using BufferCommandText in the `exec` implementation could mess up the ordering. If you did `exec myscript; echo Done`  then it would say `Done` before executing the commands in the script.